analyzers/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs

```diff
@@ -135,6 +135,13 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<Nullable<T>> enume
         public static string JoinStr(this IEnumerable<string> enumerable, string separator) =>
             JoinStr(enumerable, separator, x => x);
 
+        /// <summary>
+        /// Concatenates the members of a <see cref="string"/> collection using the specified <paramref name="separator"/> between each member.
+        /// Any whitespace or null member of the collection will be ignored.
+        /// </summary>
+        public static string JoinIfNotWhitespace(this IEnumerable<string> enumerable, string separator) =>
```

Comment:
Another idea: What about changing `JoinStr` to filter out empty stuff? It should be fine IMO as joining empty stuff doesn't generally make sense.

Reply:
I will give this a try, push and look at how many UTs/ITs are failing.

Reply:
Modifying `JoinStr` behavior leads to 189 even with ÌsNullOrEmpty` only. I will revert and use the previous approach of another method name.

Reply:
I changed the name to `JoinNonEmpty`, but there is `string.IsNullOrEmpty` and `string.IsNullOrWhitespace` so `JoinNonEmpty` is confusing, because it suggests `IsNullOrEmpty` while doing `IsNullOrWhitespace`.

Reply:
I'd say it's good enough. The whitespace isn't usually a thing that we'd like to contact with a separator.